arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,227 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
 
-```
+**Important:** Ensure your Docker service is running for this command to work correctly.
+
+```shell
 cargo stylus check
 ```
 
-This command will attempt to verify that your program can be deployed and activated onchain without requiring a transaction by specifying a JSON-RPC endpoint. See `cargo stylus check --help` for more options.
+`cargo stylus check` executes a dry-run on your project by compiling your contract to WASM and verifying if it can be deployed and activated onchain.
 
-If the command above fails, you'll see detailed information about why your WASM will be rejected:
+If the command above fails, you'll see detailed information about why your contract would be rejected:
 
-```
+```shell
 Reading WASM file at bad-export.wat
 Compressed WASM size: 55 B
-Stylus checks failed: program predeployment check failed when checking against
+Stylus checks failed: program pre-deployment check failed when checking against
 ARB_WASM_ADDRESS 0x0000…0071: (code: -32000, message: program activation failed: failed to parse program)
 
 Caused by:
     binary exports reserved symbol stylus_ink_left
 
 Location:
     prover/src/binary.rs:493:9, data: None)
-
 ```
 
-To read more about what counts as valid vs. invalid user WASM programs, see [VALID_WASM](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md). If your program succeeds, you'll see the following message:
+The program can fail the check for various reasons (on compile, deployment, etc...). Reading the [Invalid Stylus WASM Contracts explainer](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md) can help you understand what makes a WASM contract valid or not.
 
-```
+If your program succeeds, you'll see something like this:
+
+```shell
 Finished release [optimized] target(s) in 1.88s
 Reading WASM file at hello-stylus/target/wasm32-unknown-unknown/release/hello-stylus.wasm
 Compressed WASM size: 3 KB
 Program succeeded Stylus onchain activation checks with Stylus version: 1
 ```
 
-Note that running `cargo stylus check` may take a few minutes, especially on a fresh build. If you're running the check or deploy commands, a message at the start will inform you that this process might take some time. For faster checks, you can use the `--no-verify` option. Also, please ensure that Docker is up and running, as these commands are required to be executed properly.
+Note that running `cargo stylus check` may take a few minutes, especially if you're verifying a contract for the first time.
 
-Once you're ready to deploy your program onchain, you can use the `cargo stylus deploy` subcommand as follows. First, we can estimate the gas required to perform our deployment with:
+See `cargo stylus check --help` for more options.
 
-```
+## Deploying your contract
+
+Once you're ready to deploy your program onchain, `cargo stylus deploy` will help you with the estimation of the deployment's gas cost, as well as with the deployment itself.
```

Comment:
```suggestion
Once you're ready to deploy your program on-chain, `cargo stylus deploy` will help you with the estimation of the deployment's gas cost, as well as with the deployment itself.
```

Changing here since I saw you changed the previous occurrence, but feel free to use the convention you prefer.